File: java/src/PasswordManager.java
import java.io.*;
import java.util.*;

public class PasswordManager {
    private HashMap<String, String> passwords;
    private static final String FILE_PATH = "saved_passwords.txt";

    public PasswordManager() {
        passwords = new HashMap<>();
        loadFromFile();
    }

    public void savePassword(String service, String password) {
        passwords.put(service, password);
        saveToFile();
    }

    public String getPassword(String service) {
        return passwords.get(service);
    }

    public ArrayList<String> getAllServices() {
        return new ArrayList<>(passwords.keySet());
    }

    public boolean removePassword(String service) {
        String removed = passwords.remove(service);
        if (removed != null) {
            saveToFile();
            return true;
        }
        return false;
    }

    private void loadFromFile() {
        File file = new File(FILE_PATH);
        if (!file.exists()) return;
        try (BufferedReader br = new BufferedReader(new FileReader(file))) {
            String line;
            while ((line = br.readLine()) != null) {
                String[] parts = line.split(":", 2);
                if (parts.length == 2) {
                    passwords.put(parts[0], parts[1]);
                }
            }
        } catch (IOException e) {
            System.err.println("Error loading passwords: " + e.getMessage());
        }
    }

    private void saveToFile() {
        try (BufferedWriter bw = new BufferedWriter(new FileWriter(FILE_PATH))) {
            for (Map.Entry<String, String> entry : passwords.entrySet()) {
                bw.write(entry.getKey() + ":" + entry.getValue());
                bw.newLine();
            }
        } catch (IOException e) {
            System.err.println("Error saving passwords: " + e.getMessage());
        }
    }
}

File: src/security/passwordManager.js
var PasswordManager = function () {
  this._passwords = {};
  this._loadFromStorage();
};

PasswordManager.prototype._loadFromStorage = function () {
  try {
    this._passwords = JSON.parse(localStorage.getItem("savedServicePasswords") || "{}");
  } catch (e) { this._passwords = {}; }
};

PasswordManager.prototype._saveToStorage = function () {
  localStorage.setItem("savedServicePasswords", JSON.stringify(this._passwords));
};

PasswordManager.prototype.savePassword = function (service, password) {
  this._passwords[service] = password;
  this._saveToStorage();
};

PasswordManager.prototype.getPassword = function (service) {
  return this._passwords[service] || null;
};

PasswordManager.prototype.getAllServices = function () {
  return Object.keys(this._passwords);
};

PasswordManager.prototype.removePassword = function (service) {
  delete this._passwords[service];
  this._saveToStorage();
};
